core/src/main/scala/org/apache/spark/deploy/mesos/MesosClusterDispatcher.scala

```diff
@@ -50,7 +50,11 @@ private[mesos] class MesosClusterDispatcher(
   extends Logging {
 
   private val publicAddress = Option(conf.getenv("SPARK_PUBLIC_DNS")).getOrElse(args.host)
-  private val recoveryMode = conf.get("spark.mesos.deploy.recoveryMode", "NONE").toUpperCase()
+  private val recoveryMode = conf.getOption("spark.mesos.deploy.recoverMode").map { mode =>
+    logWarning("spark.mesos.deploy.recoverMode is deprecated. Please configure " +
```

Comment:
I don't think it's even worth logging a warning. It wasn't documented so any user who was using the old config somehow found out about it from the code, knowing that it's not officially supported. I'd rather keep the code simpler than add a warning that I doubt will be useful for many.